### ISSUE TITLE
Feature Request: add "pause" control to record action

### ISSUE THREAD

**Issue by yuvalif** [asciinema/asciinema#321]
When recording, sometimes you want to pause the recording, and then continue.

**Comment by just4t** [comment]
You can set an _idle_time_limit_ in a custom file _/root/.config/asciinema/config_ as follows (this would save you to record the delay times while typing your terminal entries, making the playback 'waiting' times reduced to max. 2 seconds, in this example, of course) :
```
[record]
; Limit recorded terminal inactivity to max n seconds, default: off
idle_time_limit = 2
```
Recommended [to read the docs.](https://asciinema.org/docs/config) about for more info. 

Hope this helps.

**Comment by sickill** [comment]
That's a good use case. We could implement pause/unpause via a keyboard shortcut, for ex: `<ctrl-p>` or similar.

It would probably be good to have some indicator/confirmation that you actually paused/unpaused, otherwise you would never be sure.

**Comment by vanyakosmos** [comment]
The only thing I can imagine is that recorder could write play/pause status into some file and we will need to run something like `watch -n 1 cat that_file.txt` in separate tab in order to check that status.
But this approach looks kinda ugly...

Maybe we can add this feature as post processing step:
- play recording in special "edit" mode
- first character of each row will indicate state (play/pause), maybe with some color
- on <key> press we will pause recording and change first column to mark this pause
- on second <key> press we will continue recording and change first column again
- when recording is finished we will cut off segments that were marked as pause

illustration:
```diff
+ echo recording...
+ echo still playing...
- echo this pause, it will not be present in final recording file
- echo this line too
+ echo and this line we will keep
- exit
```

**Comment by probonopd** [comment]
> You can set an idle_time_limit in a custom file /root/.config/asciinema/config 

Can this be made the default please?

**Comment by fabianbaier** [comment]
Another workaround would be:

1) Start the recording via `asciinema rec my.cast`
2) On the part where you want to **pause** press `CTRL+D` to end the recording
3) When you are ready to **resume** use `asciinema --append my.cast`
Optional: You can edit the recording and e.g. remove certain bits and pieces by editing the `my.cast` afterwards with a text-editor

Watching your recording via `asciinema play my.cast` should show you a smooth video as if you hit the pause/resume buttons.

**Comment by probonopd** [comment]
Not really, as I don't type with constant speed ;-)

**Comment by sickill** [comment]
You can "burn" fixed idle time of existing recording by playing it inside new recording:

    asciinema rec -i 2 -c "asciinema play existing.cast" new.cast

**Comment by sickill** [comment]
I've opened PR #340 for "muting" feature. Please take a look, I think this is what @yuvalif had in mind when opening this issue.

**Comment by stealthman22** [comment]
This pause feature if added will be the best. 
Sometimes I am not too sure of what I am doing on the terminal (Like reading some docs to learn something), I should be able to pause and resume easily. It's intuitive

**Comment by sickill** [comment]
This has been solved with #340, will be relased shortly with v2.1.
